proto/info/hardware.proto

```diff
@@ -0,0 +1,171 @@
+        uint64 subclass_id = 9;
+        // IOMMU group names apparently are strings: https://elixir.bootlin.com/linux/v6.16.9/source/drivers/iommu/iommu.c#L1101
+        string iommu_group = 10;
+}
+
+message SerialPort {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        string ioports = 3;
+        uint64 irq = 4;
+
+        // f.e. /dev/ttyS0
+        string devpath = 5;
+}
+
+message CANDevice {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        string ifname = 3;
+}
+
+message NetworkDevice {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        string ifname = 3;
+        bool wireless = 4;
+
+        string mac_address = 5; // e.g. "00:11:22:33:44:AA"
+        uint32 speed_mbps = 6;  // link speed if known
+}
+
+message BIOS {
+        string vendor = 1;
+        string version = 2;
+        // Curated BIOS attributes; not a full dump.
+        map<string,string> attributes = 1000;
+}
+
+message CPUCache {
+        string level = 1;      // "L1d", "L1i", "L2", "L3"
+        uint64 size_bytes = 2; // cache size in bytes
+}
+
+message CPU {
+        string model = 1;
+        string vendor = 2;          // e.g. "GenuineIntel"
+        uint32 id = 3;              // logical CPU id
+        uint64 freq = 4;             // nominal frequency
+        repeated CPUCache caches = 5;
+}
+
+message NUMA {
+        repeated CPU cpus = 1;
+}
+
+// Information about the system that is sent once when the system boots
+message ZInfoHardware {
+  // NOTE: deprecated - SMART info will be moved into ZHardwareHealth.
+  // Kept here for backward compatibility with older EVE/controller versions.
+  repeated StorageDiskInfo disks = 1; // S.M.A.R.T. info for disks
+
+  // as in /etc/eve-release
+  string eve_release = 2;
+
+  repeated PCIDevice pci_devices = 3;
+  repeated USBDevice usb_devices = 4;
+  repeated SerialPort serial_devices = 5;
+  repeated NetworkDevice network_devices = 6;
+  repeated CANDevice can_devices = 7;
+
+  BIOS bios = 8;
+  NUMA numa = 9;
+
+  // ------------------------------------------------------------------
+  // Node-level summary
+  //  - system memory
+  //  - system storage
+  //  - amount of CPUs
+  //  - availability of watchdog, TPM, status LED
+  //  - kernel flavor & version
+  //  - EVE platform
+  // ------------------------------------------------------------------
+  uint64 total_memory_bytes  = 10;  // total system RAM
+  uint64 total_storage_bytes = 11;  // total non-removable storage
+  uint32 cpu_count           = 12;  // logical CPU count
+
+  bool watchdog_present      = 13;
+  bool tpm_present           = 14;
+  bool status_led_present    = 15;
+  string kernel_flavor       = 16;  // e.g. "pc", "rt", "virt"
+  string kernel_version      = 17;  // e.g. "6.6.45-eve"
+  string eve_platform        = 18;  // board/platform name
```

Comment:
I had a previous question for this regarding Evaluation EVE's different partition. @rucoder are these fields sufficient?